.github/workflows/kernel-build.yml

```diff
@@ -0,0 +1,118 @@
+
+name: Reusable build workflow
+
+on:
+  workflow_call:
+    inputs:
+      arch:
+        required: true
+        type: string
+      toolchain_full:
+        required: true
+        type: string
+      toolchain:
+        required: true
+        type: string
+      runs_on:
+        required: true
+        type: string
+      llvm-version:
+        required: true
+        type: string
+      kernel:
+        required: true
+        type: string
+
+ 
+jobs:
+  build:
+    name: build for ${{ inputs.arch }} with ${{ inputs.toolchain_full }}
+    runs-on: ${{ fromJSON(inputs.runs_on) }}
+    timeout-minutes: 100
+    env:
+        KERNEL: ${{ inputs.kernel }}
+        REPO_ROOT: ${{ github.workspace }}
+        REPO_PATH: ""
+        KBUILD_OUTPUT: kbuild-output/
+    steps:
+      - uses: actions/checkout@v3
+        # We fetch an actual bit of history here to facilitate incremental
+        # builds (which may check out some earlier upstream change).
+        with:
+          fetch-depth: 50
+      - if: ${{ github.repository == 'kernel-patches/vmtest' }}
```

Comment:
I'd rather we make this some kind of meaningful input argument than this reliance on global state. Do you think that's possible?